From 452eac71cb88ccc3ae6362e1be8156feec7dad01 Mon Sep 17 00:00:00 2001
From: b40282 <b40282@freescale.com>
Date: Thu, 6 Dec 2012 10:04:53 +0800
Subject: [PATCH 05/14]    ENGR00235825  realtek wifi
 libhardware_legacy/wifi/Android.mk

    realtek wifi Android.mk Reference codes for applying wifi_realtek.c

    Signed-off-by: Biyong SUN <b40282@freescale.com>
---
 wifi/Android.mk | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/wifi/Android.mk b/wifi/Android.mk
index 8a206f8..ee260b0 100644
--- a/wifi/Android.mk
+++ b/wifi/Android.mk
@@ -53,9 +53,25 @@ ifeq ($(BOARD_WLAN_VENDOR),ATHEROS)
   LOCAL_C_INCLUDES += $(LOCAL_PATH)/../../external/wpa_supplicant_ath/wpa_supplicant/src/common
   LOCAL_CFLAGS += -DBOARD_HAS_ATH_WLAN=\"$(BOARD_HAS_ATH_WLAN)\"
   LOCAL_SRC_FILES += wifi/wifi_ath.c
+else ifeq ($(BOARD_WLAN_VENDOR), REALTEK)
+  LOCAL_SRC_FILES += ../realtek/wlan/libhardware_legacy/wifi/wifi_realtek.c
 else
-  LOCAL_SRC_FILES += wifi/wifi.c
-  LOCAL_C_INCLUDES += $(LOCAL_PATH)/../../external/wpa_supplicant_8/src/common
+  ifdef BOARD_HAVE_WIFI_CSR
+	LOCAL_SRC_FILES += wifi/wifi_unifi.c
+  else
+		ifeq ($(BOARD_WLAN_VENDOR),INTEL)
+			LOCAL_SRC_FILES += wifi/wifi_intel.c
+			LOCAL_C_INCLUDES += $(LOCAL_PATH)/../../external/wpa_supplicant_7/src/common
+		else
+			ifeq ($(BOARD_WLAN_VENDOR),BROADCOM)
+				LOCAL_SRC_FILES += wifi/wifi_bcm.c
+				LOCAL_C_INCLUDES += $(LOCAL_PATH)/../../external/wpa_supplicant/
+			else
+				LOCAL_SRC_FILES += wifi/wifi.c
+				LOCAL_C_INCLUDES += $(LOCAL_PATH)/../../external/wpa_supplicant_8/src/common
+			endif
+		endif
+  endif
 
 endif
 
-- 
1.8.0

